pkg/comp-functions/functions/common/maintenance.go

```diff
@@ -0,0 +1,206 @@
+package common
+
+import (
+	"context"
+	"fmt"
+	vshnv1 "github.com/vshn/appcat/apis/vshn/v1"
+	"github.com/vshn/appcat/pkg/comp-functions/runtime"
+	batchv1 "k8s.io/api/batch/v1"
+	corev1 "k8s.io/api/core/v1"
+	rbacv1 "k8s.io/api/rbac/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/utils/pointer"
+	"regexp"
+	controllerruntime "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// Maintenance contains data for maintenance k8s resource generation
+type Maintenance struct {
+	InstanceNamespace string
+	MainRole          string
+	Service           string
+	Resource          client.Object
```

Comment:
Could you add some kind of docstring explaining what all these fields are? Right now I'd have to read the code to understand what all these mean.

Reply:
Yup could catch, I will add docs all over the `maintenance.go`.